.github/workflows/solutions.yml

```diff
@@ -0,0 +1,39 @@
+name: Solutions
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "16"
+          cache: "npm"
+      - run: npm ci
+      - name: Run all solutions tests
+        run: |
+          root_dir=$(pwd)
+          for path in $(find projects/ -depth);
+            do 
+              if [ -e $path/package.json ];
+                then
+                  echo -e "\nRunning test for $path"
+                  cd $path
+                  npm run test:solutions
+                  exit_code=$?
+                  if [ $exit_code -ne 0 ];
+                    then
+                      echo "Exit code of $exit_code"
+                      exit $exit_code
+                  fi
```

Comment:
This script works in general but exits at the first failing project. That means we only get at most one project's failures logged in CI. That should be fine for most runs because most PRs will probably not impact multiple projects... but some PRs, such as dependency version upgrades and larger refactors, will.

Proposal: how about this keeps track of how many projects had a non-zero exit code? Then after running everything it can echo & exit if that count is positive.

Reply:
Perfect !

Reply:
I pushed some changes just now.. can you check ?